src/Http/Livewire/LivewireDatatable.php

```diff
@@ -627,45 +649,90 @@ public function refreshLivewireDatatable()
         $this->page = 1;
     }
 
+    public function getColumnDirection(string $sortString): string
+    {
+        $direction = 'desc';
+        if (Str::contains($sortString, '|')) {
+            $direction = Str::after($sortString, '|');
+            if ($direction !== 'asc' && $direction !== 'desc') {
+                throw new \Exception("Invalid direction $direction given in sanitizeColumnSort() method. Allowed values: asc, desc.");
+            }
+        }
+
+        return $direction;
+    }
+
     /**
      * Order the table by a given column index starting from 0.
      *
-     * @param  int  $index  which column to sort by
-     * @param  string|null  $direction  needs to be 'asc' or 'desc'. set to null to toggle the current direction.
+     * @param int $index which column to sort by
+     * @param string|null $direction needs to be 'asc' or 'desc'. set to null to toggle the current direction.
      * @return void
      */
     public function sort($index, $direction = null)
     {
-        if (! in_array($direction, [null, 'asc', 'desc'])) {
+        if (!in_array($direction, self::ORDER_BY_DIRECTION_STATES)) {
             throw new \Exception("Invalid direction $direction given in sort() method. Allowed values: asc, desc.");
         }
+        $key = Str::snake(Str::afterLast(get_called_class(), '\\'));
+
+        if ($this->multisortable) {
+            if (($valueIndexInSort = $this->getIndexesFromSort($this->sort, $index))->isEmpty()) {
```

Comment:
I think `$valueIndexInSort` is a confusing variable name. What does this variable store? We're calling `getIndexesFromSort`. I'm guessing it returns "indexes". What is an "index"? A column?